test/ParametersTest.cpp

```diff
@@ -40,3 +42,42 @@ TEST(Parameters, First) {
   ASSERT_EQ("42.000000", map.at("Float"));
   ASSERT_EQ("16.200000", map.at("Double"));
 }
+
+// Basic test, if the parameter for `MemorySize` works.
+TEST(Parameters, MemorySizeParameter) {
+  // Compare a given `MemorySizeParameter` with a given `MemorySize`.
+  auto compareWithMemorySize = [](const auto& parameter,
+                                  const MemorySize& expectedValue) {
+    ASSERT_EQ(expectedValue.getBytes(), parameter.get().getBytes());
+    ASSERT_STREQ(expectedValue.asString().c_str(),
+                 parameter.toString().c_str());
+  };
+
+  MemorySizeParameter<"Memory"> m(6_GB);
+  compareWithMemorySize(m, 6_GB);
+  m.set(6_MB);
+  compareWithMemorySize(m, 6_MB);
+  m.setFromString("6 TB");
+  compareWithMemorySize(m, 6_TB);
+
+  // Test, if it works with `Parameters`.
+  Parameters pack(Float<"Float">{2.0f}, SizeT<"SizeT">{3ull},
+                  Double<"Double">{42.1}, MemorySizeParameter<"Memory">{6_GB});
+  ASSERT_EQ(pack.get<"Memory">().getBytes(), (6_GB).getBytes());
+  pack.set("Memory", "6 MB");
+  ASSERT_EQ(pack.get<"Memory">().getBytes(), (6_MB).getBytes());
```

Comment:
Thy can't you just compare the `MemorySize` objects by equality instad of the number of bytes?j `MemorySize` should have operators like `==, <=, >` etc. (They can simply be implemented using C++20's three-way-comparison with a single function.